File: camera_backend/app/models/camera_device.py
from datetime import datetime, timezone

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class CameraDevice(Base):
    __tablename__ = "camera_devices"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    lot_id: Mapped[str] = mapped_column(String(36), ForeignKey("parking_lots.id"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    # SHA-256 hex digest of the raw api_key — never store plain key
    api_key_hash: Mapped[str] = mapped_column(String(64), nullable=False, unique=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    last_seen_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    lot = relationship("ParkingLot", back_populates="cameras")

File: lib/features/camera_management/domain/entities/camera_device.dart
import 'package:equatable/equatable.dart';

class CameraDevice extends Equatable {
  const CameraDevice({
    required this.id,
    required this.cameraUid,
    required this.name,
    required this.cameraType,
    required this.isPaired,
    required this.isActive,
    required this.createdAt,
    this.lotId,
    this.lastSeenAt,
  });

  final int id;
  final String cameraUid;
  final String? lotId;
  final String name;
  final String cameraType; // "entrance" or "exit"
  final bool isPaired;
  final bool isActive;
  final DateTime? lastSeenAt;
  final DateTime createdAt;

  @override
  List<Object?> get props => [id, cameraUid, lotId, name, cameraType, isPaired, isActive, lastSeenAt, createdAt];
}

class CameraPairResult {
  const CameraPairResult({
    required this.camera,
    required this.apiKey,
  });

  final CameraDevice camera;
  final String apiKey;
}
